catboost/libs/model/model_export/resources/apply_catboost_nan_max_model.cpp

```diff
@@ -0,0 +1,91 @@
+static int GetHash(const std::string& catFeature, const std::unordered_map<std::string, int>& catFeatureHashes) {
+    const auto keyValue = catFeatureHashes.find(catFeature);
+    if (keyValue != catFeatureHashes.end()) {
+        return keyValue->second;
+    } else {
+        return 0x7fFFffFF;
+    }
+}
+
+/* Model applicator */
+double ApplyCatboostModel(
+    const std::vector<float>& floatFeatures,
+    const std::vector<std::string>& catFeatures) {
+    const struct CatboostModel& model = CatboostModelStatic;
+
+    assert(floatFeatures.size() == model.FloatFeatureCount);
+    assert(catFeatures.size() == model.CatFeatureCount);
+
+    /* Binarize features */
+    std::vector<unsigned char> binaryFeatures(model.BinaryFeatureCount, 0);
+    unsigned int binFeatureIndex = 0;
+    {
+        /* Binarize float features */
+        for (size_t i = 0; i < model.FloatFeatureBorders.size(); ++i) {
+            if (!model.FloatFeatureBorders[i].empty()) {
+                for (const float border : model.FloatFeatureBorders[i]) {
+                    binaryFeatures[binFeatureIndex] += (unsigned char) ((floatFeatures[i] > border) || std::isnan(floatFeatures[i]));
+                }
+                ++binFeatureIndex;
+            }
+        }
+    }
```

Comment:
except one line, this file is same as apply_catboost_model.cpp, which is bad

let us call here a function, say BinarizeFloatFeatures, and provide its versions in resources/binarize_float_features.cpp and resources/binarize_float_features_nan_mode_max.cpp, so that the cat-feature part of TCatboostModelToCppConverter::Write looks like
```cpp
WriteHeader(...);
WriteBinarizer(...);
WriteModelCatFeatures(...);
WriteApplicatorCatFeatures(...);
```
And follow the same idea in python_exporter.cpp